### ISSUE TITLE
test_copy_file_prevent_dotgit_placement errored on master test of fresh git-annex

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#5248]
https://github.com/datalad/git-annex/runs/1547701854?check_suite_focus=true

```
2020-12-14T04:10:08.5210732Z ======================================================================
2020-12-14T04:10:08.5211663Z ERROR: datalad.local.tests.test_copy_file.test_copy_file_prevent_dotgit_placement
2020-12-14T04:10:08.5213262Z ----------------------------------------------------------------------
2020-12-14T04:10:08.5214090Z Traceback (most recent call last):
2020-12-14T04:10:08.5215684Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
2020-12-14T04:10:08.5216671Z     self.test(*self.arg)
2020-12-14T04:10:08.5218081Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 755, in _wrap_with_tempfile
2020-12-14T04:10:08.5219102Z     return t(*(arg + (filename,)), **kw)
2020-12-14T04:10:08.5220514Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 755, in _wrap_with_tempfile
2020-12-14T04:10:08.5221526Z     return t(*(arg + (filename,)), **kw)
2020-12-14T04:10:08.5228478Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/local/tests/test_copy_file.py", line 340, in test_copy_file_prevent_dotgit_placement
2020-12-14T04:10:08.5232082Z     dest.pathobj / 'some', '.git'], on_failure='ignore'),
2020-12-14T04:10:08.5233621Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/distribution/dataset.py", line 503, in apply_func
2020-12-14T04:10:08.5234562Z     return f(**kwargs)
2020-12-14T04:10:08.5235761Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/interface/utils.py", line 482, in eval_func
2020-12-14T04:10:08.5236766Z     return return_func(generator_func)(*args, **kwargs)
2020-12-14T04:10:08.5238059Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/interface/utils.py", line 470, in return_func
2020-12-14T04:10:08.5238961Z     results = list(results)
2020-12-14T04:10:08.5240288Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/interface/utils.py", line 401, in generator_func
2020-12-14T04:10:08.5243672Z     allkwargs):
2020-12-14T04:10:08.5245171Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/interface/utils.py", line 557, in _process_results
2020-12-14T04:10:08.5251333Z     for res in results:
2020-12-14T04:10:08.5253010Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/local/copy_file.py", line 314, in __call__
2020-12-14T04:10:08.5259119Z     message=message,
2020-12-14T04:10:08.5260420Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/distribution/dataset.py", line 503, in apply_func
2020-12-14T04:10:08.5261228Z     return f(**kwargs)
2020-12-14T04:10:08.5262267Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/interface/utils.py", line 482, in eval_func
2020-12-14T04:10:08.5263669Z     return return_func(generator_func)(*args, **kwargs)
2020-12-14T04:10:08.5264831Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/interface/utils.py", line 470, in return_func
2020-12-14T04:10:08.5265615Z     results = list(results)
2020-12-14T04:10:08.5266729Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/interface/utils.py", line 401, in generator_func
2020-12-14T04:10:08.5267498Z     allkwargs):
2020-12-14T04:10:08.5268731Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/interface/utils.py", line 557, in _process_results
2020-12-14T04:10:08.5269516Z     for res in results:
2020-12-14T04:10:08.5270487Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/core/local/save.py", line 361, in __call__
2020-12-14T04:10:08.5271180Z     lgr=lgr,
2020-12-14T04:10:08.5272145Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/support/parallel.py", line 536, in __iter__
2020-12-14T04:10:08.5272915Z     for res in super().__iter__():
2020-12-14T04:10:08.5273921Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/support/parallel.py", line 270, in __iter__
2020-12-14T04:10:08.5274712Z     yield from self._iter_serial()
2020-12-14T04:10:08.5275784Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/support/parallel.py", line 295, in _iter_serial
2020-12-14T04:10:08.5276528Z     yield from res
2020-12-14T04:10:08.5277499Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/core/local/save.py", line 301, in save_ds
2020-12-14T04:10:08.5278213Z     _status=pds_status):
2020-12-14T04:10:08.5279214Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/support/gitrepo.py", line 4161, in save_
2020-12-14T04:10:08.5279958Z     **{k: kwargs[k] for k in kwargs
2020-12-14T04:10:08.5281005Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 3338, in _save_add
2020-12-14T04:10:08.5281826Z     if expected_additions else None):
2020-12-14T04:10:08.5282980Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 1141, in _call_annex_records
2020-12-14T04:10:08.5283753Z     raise e
2020-12-14T04:10:08.5284808Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 1101, in _call_annex_records
2020-12-14T04:10:08.5285570Z     **kwargs,
2020-12-14T04:10:08.5286579Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/support/annexrepo.py", line 994, in _call_annex
2020-12-14T04:10:08.5296252Z     **kwargs)
2020-12-14T04:10:08.5297492Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/cmd.py", line 1249, in run_on_filelist_chunks
2020-12-14T04:10:08.5298203Z     **kwargs)
2020-12-14T04:10:08.5299114Z   File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/datalad/cmd.py", line 522, in run
2020-12-14T04:10:08.5299763Z     **results,
2020-12-14T04:10:08.5301723Z datalad.support.exceptions.CommandError: CommandError: 'git annex add --json --json-error-messages -c annex.dotfiles=true -c annex.retry=3 -- .git/config' failed with exitcode 1 under /tmp/datalad_temp_test_copy_file_prevent_dotgit_placementai789iv1 [err: 'error: invalid path '.git/config'
2020-12-14T04:10:08.5304156Z error: unable to add '.git/config' to index
2020-12-14T04:10:08.5304647Z fatal: adding files failed
2020-12-14T04:10:08.5305864Z git-annex: user error (xargs ["-0","git","--git-dir=.git","--work-tree=.","--literal-pathspecs","-c","annex.dotfiles=true","-c","annex.retry=3","add","--"] exited 123)'] [info keys: stdout_json]
2020-12-14T04:10:08.5307081Z -------------------- >> begin captured logging << --------------------
2020-12-14T04:10:08.5307709Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5308326Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5308919Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5309520Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5322378Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5323017Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5323610Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5324223Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5324955Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5325544Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5326148Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5326736Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5327334Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5327918Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5328518Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5329102Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5329698Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5330277Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5330863Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5331464Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5332048Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5332651Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5333237Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5333834Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5334415Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5335015Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5335597Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5336187Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5336779Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5337358Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5337952Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5338535Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5339129Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5339710Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5340294Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5340893Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5341479Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5342074Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5342909Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5343516Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5344096Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5344691Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5345274Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5345852Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5346449Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5347030Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5347629Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5348208Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5348884Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5349472Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5350066Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5350648Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5351233Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5351863Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5352447Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5353039Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5353619Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5354217Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5354799Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5355396Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5355976Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5356556Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5357154Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5357796Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5358399Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5358983Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5359584Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5360168Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5360773Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5361352Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5361934Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5362529Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5363112Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5363713Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5364298Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5364897Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5365478Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5366064Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5366658Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5367238Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5367830Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5368410Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5369012Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5369591Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5370187Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5370765Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5371344Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5371941Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5374680Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5375306Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5375894Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5376493Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5377075Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5377672Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5378256Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5378855Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5379438Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5380038Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5380618Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5381202Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5381799Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5382378Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5383131Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5383827Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5384432Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5385011Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5385592Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5386193Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5386774Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5387368Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5387955Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5388554Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5389131Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5389727Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5390310Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5390890Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5391492Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5392138Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5392733Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5393314Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5393909Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5394493Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5395088Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5395669Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5396256Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5396852Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5397438Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5398040Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5398622Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5399218Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5399804Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5400404Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5400985Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5401569Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5402165Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5402745Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5407026Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5407619Z asyncio: DEBUG: Using selector: EpollSelector
2020-12-14T04:10:08.5408620Z --------------------- >> end captured logging << ---------------------
```

**Comment by kyleam** [comment]
Hmm, no luck producing that on my local machine.  Using datalad's master (0.13.6-568-gc2a692192f), I tried with a git-annex version matching that build (8.20201127-130-g5a4aa4ae0) and git-annex's master (8.20201127-145-g5e094d02d).

**Comment by yarikoptic** [comment]
Might as well be a flaky one, the best kind ;-)

**Comment by yarikoptic** [comment]
the issue persists on github actions: https://github.com/datalad/git-annex/runs/1554553619?check_suite_focus=true and now seems to be accompanied with https://github.com/datalad/datalad/issues/5253  :-/

**Comment by kyleam** [comment]
> the issue persists on github actions

All right, thanks.  So, trying a bit harder then: if I use git 2.24 (to match that build's git), I can trigger this with datalad's master, starting with datalad's 46d78a8353 (RF: Replace all usage of AnnexRepo._run_annex_command_json(), 2020-11-18) [1].  So it looks to be an interaction between that change and the git version.  I'll dig more tomorrow.


[1] merged to master at 51d7ed056c (Merge pull request #5163 from mih/rf-callannex, 2020-12-11)

**Comment by yarikoptic** [comment]
eh, it seems we are doomed to eventually improve `datalad_installer` with `git` component and "build-from-source" method ;) neverending fun
